AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/Resolvers/Authorization/AuthorizationState+Resolver.swift

```diff
@@ -191,18 +191,28 @@ extension AuthorizationState {
                     let action = PersistCredentials(credentials: amplifyCredentials)
                     return .init(newState: .storingCredentials(amplifyCredentials),
                                  actions: [action])
-                } else if case .receivedSessionError = event.isAuthorizationEvent {
-                    // TODO: Handle session errors correctly and pass them back to the user
-                    let amplifyCredentials = AmplifyCredentials.userPoolOnly(signedInData: signedInData)
-                    let action = PersistCredentials(credentials: amplifyCredentials)
-                    return .init(newState: .storingCredentials(amplifyCredentials),
-                                 actions: [action])
-                } else if case .throwError = event.isAuthorizationEvent {
-                    // TODO: Handle session errors correctly and pass them back to the user
-                    let amplifyCredentials = AmplifyCredentials.userPoolOnly(signedInData: signedInData)
-                    let action = PersistCredentials(credentials: amplifyCredentials)
-                    return .init(newState: .storingCredentials(amplifyCredentials),
-                                 actions: [action])
+                } else if case .receivedSessionError(let fetchError) = event.isAuthorizationEvent {
+                    let amplifyCredentials = AmplifyCredentials.userPoolOnly(
+                        signedInData: signedInData)
+
+                    if case .noIdentityPool = fetchError {
+                        let action = PersistCredentials(credentials: amplifyCredentials)
+                        return .init(newState: .storingCredentials(amplifyCredentials),
+                                     actions: [action])
+                    }
+
+                    let authorizationError = AuthorizationError.sessionError(
+                        fetchError,
+                        amplifyCredentials)
+                    return .init(newState: .error(authorizationError))
+
+                } else if case .throwError(let error) = event.isAuthorizationEvent {
+                    let amplifyCredentials = AmplifyCredentials.userPoolOnly(
+                        signedInData: signedInData)
+                    let authorizationError = AuthorizationError.sessionError(
+                        .service(error),
+                        amplifyCredentials)
+                    return .init(newState: .error(authorizationError))
```

Comment:
Does this logic work when you retry? What I remember, this wasn't playing nicely when we tried to recover from a session error.

Reply:
Tested this on a sample up and retry worked as expected.